src/app/pages/mini-games/find-pokemon-name/game/components/RoundResult.jsx

```diff
@@ -0,0 +1,35 @@
+import { useCallback, useContext, useEffect } from 'react';
+import { FindPokemonNameContext } from '../state/Context';
+
+const RoundResult = () => {
+  const { state, startGame } = useContext(FindPokemonNameContext);
+  const { gameStatus } = state;
+
+  const onKeyDown = useCallback(
+    (e) => {
+      if (gameStatus !== 'win' && gameStatus !== 'lost') {
+        return;
+      }
+
+      const { code } = e;
+
+      if (code !== 'Enter' && code !== 'Space') {
+        return;
+      }
+
+      startGame();
+    },
+    [gameStatus, startGame],
+  );
+
+  useEffect(() => {
+    window.addEventListener('keydown', onKeyDown, false);
+    return () => window.removeEventListener('keydown', onKeyDown, false);
+  }, [onKeyDown]);
+
+  const buttonLabel = gameStatus === 'win' ? 'NEXT POKEMON' : 'TRY AGAIN';
```

Comment:
Add a useMeno so that only when `gameStatus` is changed calculate the label.

Reply:
Hmm I thought it was a bit overkill for a simple ternary but I've added it!

Reply:
The only reason is because of the context there are many rerenders.